src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -116,100 +114,139 @@ export class CredentialSavedObjectsClientWrapper {
 
     return {
       ...wrapperOptions.client,
-      create: createWithCredentialMaterialsEncryption,
-      bulkCreate: bulkCreateWithCredentialMaterialsEncryption,
+      create: createWithCredentialsEncryption,
+      bulkCreate: bulkCreateWithCredentialsEncryption,
       checkConflicts: wrapperOptions.client.checkConflicts,
       delete: wrapperOptions.client.delete,
       find: wrapperOptions.client.find,
       bulkGet: wrapperOptions.client.bulkGet,
       get: wrapperOptions.client.get,
-      update: updateWithCredentialMaterialsEncryption,
-      bulkUpdate: bulkUpdateWithCredentialMaterialsEncryption,
+      update: updateWithCredentialsEncryption,
+      bulkUpdate: bulkUpdateWithCredentialsEncryption,
       errors: wrapperOptions.client.errors,
       addToNamespaces: wrapperOptions.client.addToNamespaces,
       deleteFromNamespaces: wrapperOptions.client.deleteFromNamespaces,
     };
   };
 
+  private isValidUrl(endpoint: string) {
+    try {
+      return Boolean(new URL(endpoint));
+    } catch (e) {
+      return false;
+    }
+  }
+
+  private dropCredentials<T = unknown>(attributes: Omit<T, 'credentials'>) {
+    return attributes;
+  }
+
   private async validateAndEncryptAttributes<T = unknown>(attributes: T) {
     this.validateAttributes(attributes);
 
-    return await this.encryptCredentialMaterials(attributes);
+    const { noAuth } = attributes;
+
+    // Drop credentials when no Auth
+    if (!noAuth) {
+      return this.dropCredentials(attributes);
+    }
+
+    const { type, credentialsContent } = attributes.credentials;
+
+    switch (type) {
```

Comment:
How could this be executed, if line 151 returned, given `noAuth = false`.

Reply:
the model changed, let's review the most updated code